src/modules/ims_registrar_pcscf/save.c

```diff
@@ -467,6 +484,20 @@ int save(struct sip_msg* _m, udomain_t* _d, int _cflags) {
 		goto error;
 	}
 
+        // skip subscribe, if all contacts have "expires=0" parameter
+	contact_t *ctct;
```

Comment:
Move the contact and also expir parameter definition to the beginning of the function. Not sure why the cast to unsigned int * is needed here? The if (expir) condition also need to be aligned from the formatting.

Reply:
Moved the new definitions to the beginning of the function and did some formatting